src/materializer_vnode.erl

```diff
@@ -634,86 +594,51 @@ check_filter(Fun, Id, Last, NewId, Tuple, NewSize, NewOps) ->
             check_filter(Fun, Id+1, Last, NewId, Tuple, NewSize, NewOps)
     end.
 
-%% This is an internal function used to convert the tuple stored in ets
-%% to a tuple and list usable by the materializer
-%% Note that the ops are stored in the ets with the most recent op at the end of
+%% @doc Extract from the tuple stored in the operation cache
+%% 1) the key, 2) length of the op list (stored in form of a tuple),
+%% 3) sequence number of operation (used to trigger GC regularly),
+%% 4) size of the tuple that contains the op list, 5) the tuple containing the op list
+%% Note that the ops in the tuple are stored in the ets with the most recent op at the end of
 %% the tuple.
-%% The second argument if true will convert the ops tuple to a list of ops
-%% Otherwise it will be kept as a tuple
--spec tuple_to_key(tuple(), boolean()) -> {any(), integer(), non_neg_integer(), non_neg_integer(),
-                      [op_and_id()]|tuple()}.
-tuple_to_key(Tuple, ToList) ->
+-spec deconstruct_opscache_entry(tuple()) ->
+    {key(), non_neg_integer(), non_neg_integer(), non_neg_integer(), [op_and_id()] | tuple()}.
```

Comment:
The last component of the return type should be just `tuple()` instead of `[op_and_id()] | tuple()`. `[op_and_id()]` was only for the `true` case in the old code.

Probably this change also affects the types of `fetch_updates_from_cache` and the `ops_list` field in the `snapshot_get_response` record.

Maybe it would be clearer to write getters for the 4 parts?

Reply:
I am hesitant to have more changes on materializer_vnode as this will be replaced anyway.